modules/Microsoft.Cache/redis/deploy.bicep

```diff
@@ -0,0 +1,262 @@
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. Enables system assigned managed identity on the resource.')
+param systemAssignedIdentity bool = false
+
+@description('Optional. The ID(s) to assign to the resource.')
+param userAssignedIdentities object = {}
+
+@description('Optional. Specifies whether the non-ssl Redis server port (6379) is enabled.')
+param enableNonSslPort bool = false
+
+@allowed([
+  '1.0'
+  '1.1'
+  '1.2'
+])
+@description('Optional. Specify the type of lock.')
+param minimumTlsVersion string = '1.2'
+
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+@description('Optional. Control permission for data plane traffic coming from public networks while private endpoint is enabled.')
+param publicNetworkAccess string = 'Enabled'
+
+@description('Optional. All Redis Settings. Few possible keys: rdb-backup-enabled,rdb-storage-connection-string,rdb-backup-frequency,maxmemory-delta,maxmemory-policy,notify-keyspace-events,maxmemory-samples,slowlog-log-slower-than,slowlog-max-len,list-max-ziplist-entries,list-max-ziplist-value,hash-max-ziplist-entries,hash-max-ziplist-value,set-max-intset-entries,zset-max-ziplist-entries,zset-max-ziplist-value etc.')
+param redisConfiguration object = {}
+
+@allowed([
+  '4'
+  '6'
+])
+@description('Optional. Redis version. Only major version will be used in PUT/PATCH request with current valid values: (4, 6).')
+param redisVersion string = '6'
+
+@minValue(1)
+@description('Optional. The number of replicas to be created per primary.')
+param replicasPerMaster int = 1
+
+@minValue(1)
+@description('Optional. The number of replicas to be created per primary.')
+param replicasPerPrimary int = 1
+
+@minValue(1)
+@description('Optional. The number of shards to be created on a Premium Cluster Cache.')
+param shardCount int = 1
+
+@allowed([
+  0
+  1
+  2
+  3
+  4
+  5
+  6
+])
+@description('Optional. The size of the Redis cache to deploy. Valid values: for C (Basic/Standard) family (0, 1, 2, 3, 4, 5, 6), for P (Premium) family (1, 2, 3, 4).')
+param capacity int = 1
+
+@allowed([
+  'C'
+  'P'
+])
+@description('Optional. The SKU family to use. (C = Basic/Standard, P = Premium).')
+param family string = 'P'
+
+@allowed([
+  'Basic'
+  'Premium'
+  'Standard'
+])
+@description('Optional. The type of Redis cache to deploy.')
+param skuName string = 'Premium'
```

Comment:
Should `Premium` be the default sku? Wondering about the additional costs wrt `Basic` or `Standard`

Reply:
@MrMCake explained that by the principle of secure defaults, the default SKU should always be one that doesn't limit security relevant features such as managed identities, public endpoints, etc..

Reply:
Hey @itpropro, I think you're referring to my comment on the Web PubSub module where even the security policy created an Audit if one uses the Free-Tier. For that one the reason was (I believe) that one can only deploy Private Endpoints using the Standard Tier.
Now regarding the Redis Cache module we have to again make a case by case evaluation. @itpropro , are you aware of any specific security features (like private endpoints) we can only enable on the premium plan? I will also try and check the policies...

Reply:
Ok, so I wasn't able to find anything SKU related in any policy recommendations (e.g. [here](https://github1s.com/Azure/azure-policy/blob/master/built-in-policies/policyDefinitions/Cache/RedisCache_AuditSSLPort_Audit.json)). Further, I just clicked through the service creation and it seems you can create e.g. private endpoints with the Standard SKU without any troubles.